app/uk/gov/hmrc/transitmovements/controllers/MovementsController.scala

```diff
@@ -135,6 +142,62 @@ class MovementsController @Inject() (
       case None    => updateMovementLargeMessage(movementId, triggerId)
     }
 
+  def updateMessage(movementId: MovementId, messageId: MessageId) =
+    Action.async(parse.json) {
+      implicit request =>
+        mapRequest(request.body)
+          .flatMap {
+            largeMessageMetadata => handleUpdateMessage(largeMessageMetadata, request, movementId, messageId)
+          }
+          .fold[Result](
+            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
+            x => x
+          )
+    }
+
+  private def handleUpdateMessage(largeMessageMetadata: LargeMessageMetadata, request: Request[JsValue], movementId: MovementId, messageId: MessageId)(implicit
+    hc: HeaderCarrier
+  ) =
+    largeMessageMetadata match {
+      case LargeMessageMetadata(Some(_), status) =>
+        (for {
+          messageType <- extract(request.headers).asPresentation
+          sourceFile  <- objectStoreService.getObjectStoreFile(largeMessageMetadata.objectStoreURI.get).asPresentation
+          messageData <- messagesXmlParsingService.extractMessageData(sourceFile, messageType).asPresentation
+          received = OffsetDateTime.ofInstant(clock.instant, ZoneOffset.UTC)
+          message = messageFactory.updateLargeMessage(
+            messageType,
+            Some(messageData.generationDate),
+            received,
+            messageId,
+            largeMessageMetadata.objectStoreURI,
+            status
+          )
+          _ <- repo.updateMessages(movementId, message, messageData.mrn, received).asPresentation
```

Comment:
We need a new function to update the array field in the message. This function adds a message to the movement rather than updates fields in a message in a movement. It should be renamed to something like addMessage and a new function created for the updating.

Reply:
added a new method for updateMessage